Modules/_ctypes/cfield.c

```diff
@@ -144,11 +129,69 @@ PyCField_new_impl(PyTypeObject *type, PyObject *name, PyObject *proto,
                              ((PyTypeObject*)proto)->tp_name);
                 goto error;
         }
+
+        if (byte_size > 100) {
+            // Bitfields must "live" in a field defined by a ffi type,
+            // so they're limited to about 8 bytes.
+            // This check is here to avoid overflow in later checks.
+            PyErr_Format(PyExc_ValueError,
+                        "bit field %R size too large, got %zd",
+                        name, byte_size);
+        }
+        bitfield_size = PyLong_AsSsize_t(bit_size_obj);
+        if ((bitfield_size <= 0) || (bitfield_size > 255)) {
+            if (!PyErr_Occurred()) {
+                PyErr_Format(PyExc_ValueError,
+                            "bit size of field %R out of range, got %zd",
+                            name, bitfield_size);
+            }
+            goto error;
+        }
+        bit_offset = PyLong_AsSsize_t(bit_offset_obj);
+        if ((bit_offset < 0) || (bit_offset > 255)) {
+            if (!PyErr_Occurred()) {
+                PyErr_Format(PyExc_ValueError,
+                            "bit offset of field %R out of range, got %zd",
+                            name, bit_offset);
+            }
+            goto error;
+        }
+        if ((bitfield_size + bit_offset) > byte_size * 8) {
+            PyErr_Format(
+                PyExc_ValueError,
+                "bit field %R overflows its type (%zd + %zd >= %zd)",
+                name, bit_offset, byte_size*8);
+                goto error;
+        }
+    }
+    else {
+        if (bit_offset_obj != Py_None) {
+            PyErr_Format(
+                PyExc_ValueError,
+                "field %R: bit_offset must be specified if bit_size is",
+                name);
+        }
+    }
+
+    self = _CFieldObject_CAST(type->tp_alloc(type, 0));
+    if (!self) {
+        return NULL;
+    }
+    if (PyUnicode_CheckExact(name)) {
+        self->name = Py_NewRef(name);
+    } else {
+        self->name = PyObject_Str(name);
```

Comment:
IIRC, in `CFieldObject`, `name` is documented as being an *exact* unicode object but this is not necessarily the case as `PyObject_Str` may return a non-exact unicode object.

Reply:
Thanks for catching these!